### PR TITLE
doc/md/guides: my102 guide

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/md/guides/mysql/my-102.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This document describes in detail the detection of inline `REFERENCES` clauses
+for MySQL.
+
+According to the [MySQL Documentation](https://dev.mysql.com/doc/refman/8.0/en/create-table.html):
+> MySQL parses but ignores "inline REFERENCES specifications" 
+> (as defined in the SQL standard) where the references are 
+> defined as part of the column specification
+
+This means, that despite the fact that in many databases users can define
```
Comment (hilakashai) [inline doc/md/guides/mysql/my-102.md:18]:
```suggestion
This means that despite the fact that in many databases users can define
```

File: doc/md/guides/mysql/my-102.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    owner_id int REFERENCES users(id)
+);
+```
+The database will silently ignore the `REFERENCES users(id)` clause.
+
+## Impact
+
+### Referential integrity
+
+REFERENCES clauses are used to define referential constrains (foreign-keys)
```
Comment (hilakashai) [inline doc/md/guides/mysql/my-102.md:33]:
```suggestion
REFERENCES clauses are used to define referential constrains (foreign keys)
```

File: doc/md/guides/mysql/my-102.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+that allow application developers to rely on the database to maintain referential integrity.
+
+A developer that successfully applied the `CREATE TABLE` statement above
+may wrongfully believe that the database will reject any inserts of 
+records that refer to non-existing records. As a result, other means of
+validation may not be implemented which may lead to a situation where
```
Comment (hilakashai) [inline doc/md/guides/mysql/my-102.md:39]:
```suggestion
validation may not be implemented, which may lead to a situation where 
```

File: doc/md/guides/mysql/my-102.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+Preventing migrations with inline REFERENCES clauses is easy with Atlas's [`migrate lint`](/versioned/lint)
+command. With the `atlas migrate lint` command, users can analyze the migration directory to
+detect potentially dangerous changes to the database schema. This command may be
+incorporated in continuous integration pipelines to enable teams to enforce
+desired policies with regard to schema changes.
+
+When using migrate lint to analyze migrations, users must supply multiple parameters:
+
+* `--dev-url` - a URL to a Dev-database that will be used to simulate the changes and verify their correctness.
```
Comment (hilakashai) [inline doc/md/guides/mysql/my-102.md:54]: Link to dev db?

**Review by hilakashai — APPROVED** [review]
🚀

**Review by a8m — APPROVED** [review]

**Pull request merged**